pallets/stableswap/src/trade_execution.rs

```diff
@@ -49,7 +144,25 @@ impl<T: Config> TradeExecution<T::RuntimeOrigin, T::AccountId, T::AssetId, Balan
 	) -> Result<(), ExecutorError<Self::Error>> {
 		match pool_type {
 			PoolType::Stableswap(pool_id) => {
-				Self::sell(who, pool_id, asset_in, asset_out, amount_in, min_limit).map_err(ExecutorError::Error)
+				if asset_in == pool_id {
+					Self::remove_liquidity_one_asset(who, pool_id, asset_out, amount_in, min_limit)
+						.map_err(ExecutorError::Error)
+				} else if asset_out == pool_id {
+					let decimals = T::AssetInspection::decimals(asset_in)
```

Comment:
do you really have to specify decimals in extrinsic?

Reply:
removed